changelogs/fragments/6192-allow-empty-resultsets.yml

```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - mssql_script_module - Handle error condition for empty resultsets to allow for non-returning SQL statements (eg. UPDATE/INSERT)
+  - mssql_script_module - Allow for "GO" statement to be mixed-case for scripts not using strict syntax
```

Comment:
```suggestion
  - mssql_script - handle error condition for empty resultsets to allow for non-returning SQL statements (for example ``UPDATE`` and ``INSERT``) (https://github.com/ansible-collections/community.general/pull/6457).
  - mssql_script - allow for ``GO`` statement to be mixed-case for scripts not using strict syntax (https://github.com/ansible-collections/community.general/pull/6457).
```

Reply:
@felixfontein I have applied your suggestions as well as updated the batching logic to address some issues I discovered during testing with pulling (slurp) scripts from Windows machines. The logic is more complicated but, I believe, more reliably batches scripts from a wider array of sources.